### ISSUE TITLE
Switch to Preset mode does not work in dialog box

### ISSUE THREAD

**Issue by Gyosa3** [msvisser/remeha_home#11]
Hi,

I was testing the change between modes in HA to see how the app was reacting. I have spotted an issue that I find a bit weird, I can't say if it's my setup that has a problem in particular or not.

When I click on the Lovelace "Thermostat" card, I can very well switch between all 3 modes by clicking on the 3 icons. Moreover when I switch from "off" or "heat" to "schedule", the schedule preset applied is the last one used, as expected. So all fine until there.

**Issue part 1:**
So I am in "Schedule" mode with "Clock program 1" active. Then if I want to change the schedule program, I need to click on the 3 vertical dots and select a new preset in the drop down. When I do this, the preset is applied and the set temperature is changed both in the integration and in the app, but the app still indicates "Heating 1" on the top right, and "Schedule: Heating 1 active" on the main screen. So it seems that another schedule is followed as expected but the app has an issue with displaying it properly. (it works OK when the preset is changed from the app directly). I think for this part of the issue it's a bug in the app itself.

**Issue part 2:**
Now I change to "Heat" or "Off" modes in the integration and I let the app sync correctly. I click the 3 dots and directly select a preset, which should automatically switch the mode to "schedule" thanks to this code:
```
        self._data["zoneMode"] = HVAC_MODE_TO_REMEHA_MODE.get(HVACMode.AUTO)
        self._data["activeHeatingClimateTimeProgramNumber"] = target_preset
        self.async_write_ha_state()
```
It does try to switch to schedule mode for a while but then reverts to "Heat" or "Off" systematically when it resync. So in this particular configuration the "double command" (change mode + change preset) is not sent to the API properly as it seems, or equally the API has an issue with its processing.

What I noticed is that for the change of mode this function is called:
```
            await self.api.async_set_schedule(
                self.climate_zone_id,
                self._data["activeHeatingClimateTimeProgramNumber"],
            )
```
while when directly selecting a preset, this function is called:
```
        await self.api.async_activate_heating_time_program(
            self.climate_zone_id, target_preset
        )
```
I don't know the difference between the two calls, but one seems to work better than the other, in case the mode is not already "schedule".

![image](https://user-images.githubusercontent.com/51777889/220391664-f4364bac-760d-43db-ab65-e33f5e12ad17.png)

**Comment by msvisser** [comment]
Thanks for this very thorough bug report.

It seems I misunderstood what the API calls are doing exactly, so when setting the preset, the mode is not actually changed to auto/clock program. The first part of code that you identified is a bit of hack to make sure the HA interface updates immediately after you change the setting, before the cloud actually has time to update and inform us of the change. Here I made the mistake of changing the "zoneMode" while the API call you identified in the third snippet does not do this.

There are two options to solve this. I can either not update the mode, which matches the app does when you change the clock program while in manual mode. Or I can automatically change the mode to auto when you select a preset. Let me know what you prefer.

**Comment by Gyosa3** [comment]
I've just tested back on the app, you're right, changing the program does not switch to "schedule" mode. Not sure if it's really intended though considering the other bug I reported above...

Personally, if you really ask me, I'd find convenient to have the mode changed to "auto" when you change the preset. If the integration does this, I fully rely on you ;) to do the sync of 2 consecutive changes (auto mode and preset) for me. Doing that in an automation is possible, but feels less reliable to throw 2 service calls to the same entity one after the other. It's just a personal opinion here.

I believe that this should also be documented in your install page, so people will be aware of this behaviour.

PS: I have no problem with the hack of displaying the target state before it is really synced. Other integrations do this as well, and yet others ask if you want to pre-display the target state before the sync or if you want to wait for the sync to have happened to display the target state. If the API is slow but reliable, as I expect the Remeha is, then anticipating the change with your hack should be a better option, I agree with you.
